sql/core/pom.xml

```diff
@@ -223,6 +223,18 @@
       <artifactId>htmlunit3-driver</artifactId>
       <scope>test</scope>
     </dependency>
+    <!-- Explicit declaration of bouncy-castle dependencies are
+         needed for maven test builds on later hadoop releases.-->
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcprov-jdk18on</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcpkix-jdk18on</artifactId>
+      <scope>test</scope>
+    </dependency>
```

Comment:
Could you clean up the existing YARN module dependency in this PR?

https://github.com/apache/spark/blob/7e7ba4eaf0716676bc914706d8f6e2a61ee8b2c4/resource-managers/yarn/pom.xml#L54-L64